test/e2e/ipsec_test.go

```diff
@@ -43,3 +48,65 @@ func TestIPSecTunnelConnectivity(t *testing.T) {
 	// Restore normal Antrea deployment with IPSec disabled.
 	data.redeployAntrea(t, false)
 }
+
+// TestIPSecDeleteStaleTunnelPorts checks that when switching from IPsec mode to
+// non-encrypted mode, the previously created tunnel ports are deleted
+// correctly.
+func TestIPSecDeleteStaleTunnelPorts(t *testing.T) {
+	if testOptions.providerName == "kind" {
+		t.Skipf("Skipping test for the KIND provider as IPSec tunnel does not work with KIND")
+	}
+	if clusterInfo.numNodes < 2 {
+		t.Skipf("Skipping test as it requires 2 different nodes")
+	}
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	t.Logf("Redeploy Antrea with IPSec tunnel enabled")
+	data.redeployAntrea(t, true)
```

Comment:
How long it takes to redeploy Antrea? I wonder if the e2e tests will last too long if we continue to add such tests.

Reply:
After the fix to the `start_ovs_ipsec` it went from 80s to 30s. I think as long as the tests take about the same amount of time as the conformance / NP tests, we are good. We still have a few minutes of margin I believe.